contracts/Vault1.sol

```diff
@@ -1,6 +1,37 @@
 //SPDX-License-Identifier: Unlicense
 pragma solidity ^0.8.4;
 
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "hardhat/console.sol";
+import "./TestERC.sol";
+
 contract Vault1 {
-    
+    address owner;
+    mapping(address => uint256) userBalance;
+
+    constructor() {
+        owner = msg.sender;
+    }
+
+    function deposit(uint256 _amount, address _tokenType) external {
+        TestERC token = TestERC(_tokenType);
```

Comment:
If its always one kind of token i.e., TestERC it is better to provide its address as parameter to constructor